keeper/src/main/scala/zio/keeper/membership/swim/Protocol.scala

```diff
@@ -0,0 +1,115 @@
+package zio.keeper.membership.swim
+
+import zio.keeper.{ Error, TaggedCodec }
+import zio.logging.Logging
+import zio.stream.ZStream
+import zio.{ Chunk, Queue, ZIO }
+
+/**
+ * Protocol represents message flow.
+ * @tparam A - Address type of sender and receiver
+ * @tparam M - type of messages handles by this protocol
+ */
+trait Protocol[A, M] {
+  self =>
+
+  /**
+   * Converts this protocol to Chunk[Byte] protocol. This helps with composing multi protocols together.
```

Comment:
Does it make sense to expose only binary for now?

Reply:
It might be handy to implement messages that we need to piggy backed. You can create protocol and then filter what you want to piggyback and then convert to binary protocol to be able to compose.